Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule_test.py

```diff
@@ -2287,3 +2273,24 @@ def test_create_x509_certificate_object():
         "issuer": "C=ZA, ST=Western Cape, L=Cape Town, O=Thawte Consulting cc, OU=Certification Services Division,"
         " CN=Thawte Server CA/emailAddress=server-certs@thawte.com",
     }
+
+
+@pytest.mark.parametrize(
+    "pattern, value",
+    [
+        ("[domain-name:value = 'www.example.com']", 'www.example.com'),
+        ("[file:hashes.'SHA-256' = '0000000000000000000000000000000000000000000000000000000000000000']",
+         '0000000000000000000000000000000000000000000000000000000000000000'),
+        ("[file:hashes.'MD5' = '00000000000000000000000000000000']", '00000000000000000000000000000000')
+    ],
+)
+def test_get_single_pattern_value(pattern, value):
```

Comment:
It doesn't support complex patterns. I spoke to @Ni-Knight and we're not going to support this case.

Reply:
I understand but the current implementation will retrieve the latest name in case we are getting a complex pattern so is this the desired behavior or should we retrieve the first one?

Reply:
Added a complex pattern to the test cases.